pkg/securitypolicy/securitypolicyenforcer_rego.go

```diff
@@ -277,17 +287,68 @@ func (policy *regoEnforcer) EnforceDeviceMountPolicy(target string, deviceHash s
 		return fmt.Errorf("device mount not allowed by policy.\ninput: %s", string(inputJSON))
 	}
 
-	deviceMap := policy.data["devices"].(map[string]string)
-	if _, ok := deviceMap[target]; ok {
-		inputJSON, err := json.Marshal(input)
-		if err != nil {
-			return fmt.Errorf("unable to marshal the Rego input data: %w", err)
+	err = policy.updateMetadata(results)
+	if err != nil {
+		return fmt.Errorf("unable to update metadata: %w", err)
+	}
+
+	return nil
+}
+
+func (policy *regoEnforcer) updateMetadata(results map[string]interface{}) error {
+	metadata := policy.data["metadata"].(map[string]map[string]interface{})
+	for name, value := range results {
+		if name == "allowed" {
+			continue
+		}
+
+		if _, ok := metadata[name]; !ok {
+			metadata[name] = make(map[string]interface{})
 		}
 
-		return fmt.Errorf("device %s already mounted.\ninput: %s", target, string(inputJSON))
+		data, ok := value.(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("unable to load metadata object")
+		}
+		action, ok := data["action"].(string)
+		if !ok {
+			return fmt.Errorf("unable to load metadata action")
+		}
+		key, ok := data["key"].(string)
+		if !ok {
+			return fmt.Errorf("unable to load metadata key")
+		}
+
+		switch action {
+		case "add":
+			_, ok := metadata[name][key]
+			if ok {
+				return fmt.Errorf("cannot add metadata value, key already exists")
```

Comment:
it's probably worth returning `name` and `key`

Reply:
Agreed, I'll make this change.